### PR TITLE
Ascii subsequences

### PR DESCRIPTION
**Pull request by Codess08**
Ascii Subsequences Problem java solution

Issue No : #1677

Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Languages Used:
- Java

#### Files Added:
- Ascii_Subsequences.java

### CODE REVIEW

**Comment by Bhavyaa08** [comment]
And also I am sorry I just saw, I will work only on one issue per PR from the next time.

**Comment by jainaman224** [comment]
Create different PR for different issues.

File: Is_Binary_Search_Tree/isBST.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Given a binary tree we have to tell whether it is a binary search tree as well or not
+
+import java.util.Scanner;
+
+public class BinaryTree {
+
+    class Node {
+	int data;
```
Comment (jainaman224) [inline Is_Binary_Search_Tree/isBST.java]: Use 4 space indentation.

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public static void main(String args[]) {
+    // Your Code Here
+	Scanner obj = new Scanner(System.in);
+	String str = obj.next();
+	int ans = find(str, ""); //this function will return count of such subsequences and also print them
+	System.out.println();
+	System.out.println(ans); //this is to print the count
+}
+
+    public static int find(String str, String ans) {
```
Comment (jainaman224) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Incorrect indentation.

**Comment by Bhavyaa08** [comment]
Please check now if the indentation is correct or not @jainaman224

**Comment by Bhavyaa08** [comment]
And I have deleted the isBST file from this branch, and am creating a new PR for it.

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+    /* Take as input str, a string. We are concerned with all the possible ascii-subsequences of str.
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Why have you added 1-level indentation by default in the whole file??

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    import java.util.Scanner;
+
+    public class Ascii_Subsequences {
+        public static void main(String args[]) {
+	    // Your Code Here
+	    Scanner obj = new Scanner(System.in);
+	    String str = obj.next();
+	    int ans = find(str, "");
+	    //this function will return count of such subsequences and also print them
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Add a space after //

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	    // when ch's ascii code is added to the ans
+
+	    return x + y + z; 
+            //the count for this frame will be sum of x,y,z
+        }
+   }
+
+    // Sample input : ab
+    // Sample output :
+    //  b 98 a ab a98 97 97b 9798
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Only 1 space should be there after //

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	    }
+
+	    char ch = str.charAt(0);
+	    int x = find(str.substring(1), ans); // when ans is passed as it is
+	    int y = find(str.substring(1), ans + ch); // when ch is added to the ans
+	    int z = find(str.substring(1), ans + (int) ch);
+	    // when ch's ascii code is added to the ans
+
+	    return x + y + z; 
+            //the count for this frame will be sum of x,y,z
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Add space after // and after comma

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        }
+   }
+
+    // Sample input : ab
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Remove space before :

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java (hunk 5 above)
Comment (Bhavyaa08) [reply ASCII_SUBSEQUENCES/Ascii_subsequences.java]: the first output of the input ab is a blank string i.e. a space. That's why an extra space is being shown in the output.

**Comment by SakshiShreya** [comment]
Please resolve the comment by jainaman224.

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public static int find(String str, String ans) {
+        if (str.length() == 0) {
+	    System.out.print(ans + " ");
+	    return 1; // This will develop the count of such subsequences
+	}
+
+	char ch = str.charAt(0);
+	int x = find(str.substring(1) , ans); // when ans is passed as it is
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Remove the space before comma

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+	int y = find(str.substring(1) , ans + ch); // when ch is added to the ans
+	int z = find(str.substring(1), ans + (int) ch);
+	// when ch's ascii code is added to the ans
+
+	return x + y + z; 
+        // The count for this frame will be sum of x, y, z
+   }
+}
+
+    // Sample input : ab
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Remove the indentation here.

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
+ /* Take as input str, a string. We are concerned with all the possible ascii-subsequences of str.
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java:1]: Please remove the space before starting the comment.

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    Sample input : ab
+    Sample output :
+     b 98 a ab a98 97 97b 9798
+     9 */
+
+import java.util.Scanner;
+
+public class Ascii_Subsequences {
+    public static void main(String args[]) {
+	Scanner obj = new Scanner(System.in);
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java:12]: Indentation is not correct. There is a mix of 4 spaces and 1 tab in the whole file.

**Comment by Codess08** [comment]
Please check all the three updated files. And since I am having trouble in the inundation part, please guide me through so that I can work on more issues.

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java (hunk 11 above)
Comment (Codess08) [reply ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Isnt 1 tab = 4 spaces?

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+public class Ascii_Subsequences {
+	
+    public static void main(String args[]) {
+	   
+        /* In this problem we have to find all the possible Ascii subsequences
+         * of the given string */
+    
+        Scanner obj = new Scanner(System.in);
+	String str = obj.next();
```
Comment (somya-kapoor) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java:11]: inconsistency in indentation

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+	System.out.println(ans); 
+	
+    }
+
+    public static int find(String str, String ans) {
+	    
+        if (str.length() == 0) {
+            
+            System.out.print(ans + " ");
+	        return 1;
```
Comment (somya-kapoor) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java:25]: 4 space indentation

Comment (Codess08) [reply ASCII_SUBSEQUENCES/Ascii_subsequences.java:25]: image

Comment (Codess08) [reply ASCII_SUBSEQUENCES/Ascii_subsequences.java:25]: What is the problem in this @somya-kapoor

Comment (somya-kapoor) [reply ASCII_SUBSEQUENCES/Ascii_subsequences.java:25]: Screenshot (122)

Comment (somya-kapoor) [reply ASCII_SUBSEQUENCES/Ascii_subsequences.java:25]: it looks like 2 space indentation

**Comment by MastersAbh** [comment]
Are you working on it @Codess08 ?

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+	   
+        /* In this problem we have to find all the possible Ascii subsequences
+         * of the given string */
+    
+        Scanner obj = new Scanner(System.in);
+	String str = obj.next();
+	    
+	int ans = find(str, ""); 
+	
+	System.out.println();
```
Comment (MastersAbh) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java:15]: we dont need this line 15

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 15]
```diff
@@ -0,0 +1,42 @@
+import java.util.Scanner;
```
Comment (MastersAbh) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java:1]: Please add problem statement at top of file

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by MastersAbh** [comment]
I see no update from your side for over a month @Codess08 so closing the PR. reach out before end of deadline if you want to work on this.

**Pull request closed without merging**